src/olympia/amo/utils.py

```diff
@@ -1161,10 +1161,10 @@ def extract_colors_from_image(path):
     return colors
 
 
-def use_fake_fxa():
+def use_fake_fxa(config):
     """Return whether or not to use a fake FxA server for authentication.
     Should always return False in production"""
-    return settings.DEV_MODE and settings.USE_FAKE_FXA_AUTH
+    return config.get('client_id') == '.'
```

Comment:
The main benefit of the explicit settings checks was to avoid accidentally having fake FxA enabled on dev/stage/prod if an environnement variable wasn't set properly (something we can't easily see/test/control ourselves until after deployment).

We should at least keep something like `USE_FAKE_FXA_AUTH`, change it `ALLOW_FAKE_FXA_AUTH` and use it here on top of the check on `client_id` being `.` to keep the same guarantee.

Reply:
I was thinking about a Django system check that raises if those values are not set in production. Wdyt?

Reply:
I was actually thinking we could have a django check that ensures those values are not '.' in production. Would that also solve the issue? Production pods would just fail if they start without proper auth configured.

Reply:
I found a potential compromise. we can re-use the existing `ENV` setting to check if we are running on a "local" environment or not and only allow fxa if both ENV == 'local' and FXA_CONFIG is as expected.

I've had bad experiences previously trying to add "custom" flags in settings. I think we should aim for a narrow waist and have a setting that tells us the information we actually care about "are you in prod or not" in a way that is clear and doesn't have any explicit meaning about how that information should be used.

Otherwise we end up having many of these bespoke flags that all should agree with each other but then become impossible to coordinate correctly.